lib/lightning_network/invoice.ex

```diff
@@ -581,9 +583,15 @@ defmodule Bitcoinex.LightningNetwork.Invoice do
     case result do
       {:ok, amount_in_bitcoin} ->
         amount_msat_dec = D.mult(amount_in_bitcoin, @milli_satoshi_per_bitcoin)
-        amount_msat = D.to_integer(amount_msat_dec)
+        rounded_amount_msat_dec = D.round(amount_msat_dec)
 
-        {:ok, amount_msat}
+        case Decimal.equal?(rounded_amount_msat_dec, amount_msat_dec) do
```

Comment:
Why use `Decimal` instead of `D`?

Reply:
I changed it back to using `D` to be more consistent. Thanks!